packages/block-editor/src/components/inserter/menu.native.js

```diff
@@ -178,6 +177,13 @@ function InserterMenu( {
 		setShowTabs,
 	] );
 
+	function isFullScreen() {
+		// The bottomsheet will not re-render after setting showSeachForm to false
+		// after mounting the component. In this case the searchForm is fliped from true to false
+		// after checking the min items for search but we can make that check on the first render.
+		return ! isIOS && showSearchForm && items.length > MIN_ITEMS_FOR_SEARCH;
```

Comment:
I noticed that `showSearchForm` is correctly updated after the first render ( via the `useEffect` hook above). However the `BottomSheet` is not set up to update when props change. 

The alternative to this change is to have the `BottomSheet` listen for prop changes but that is not recommended and this route saves an extra render